javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -2,173 +2,176 @@ package java.util
-    fillImpl(a, fromIndex, toIndex, value)
+  def sort(a: Array[Float]): Unit =
+    sortImpl(a)
 
-  def fill(a: Array[Float], value: Float): Unit =
-    fillImpl(a, value)
+  def sort(a: Array[Float], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Float](a, fromIndex, toIndex, new Array(_))
 
-  def fill(a: Array[Float], fromIndex: Int, toIndex: Int, value: Float): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  def sort(a: Array[Double]): Unit =
+    sortImpl(a)
 
-  def fill(a: Array[Double], value: Double): Unit =
-    fillImpl(a, value)
+  def sort(a: Array[Double], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Double](a, fromIndex, toIndex, new Array(_))
 
-  def fill(a: Array[Double], fromIndex: Int, toIndex: Int, value: Double): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  def sort(a: Array[AnyRef]): Unit =
+    sortImpl(a)
 
-  private def fillImpl[@specialized T](a: Array[T], value: T): Unit = {
-    var i = 0
-    while (i != a.length) {
-      a(i) = value
-      i += 1
-    }
+  def sort(a: Array[AnyRef], fromIndex: Int, toIndex: Int): Unit = {
+    sortRangeImpl[AnyRef](a, fromIndex, toIndex,
+      java.lang.reflect.Array.newInstance(
+        a.getClass.getComponentType, _).asInstanceOf[Array[AnyRef]])
   }
 
-  private def fillImpl[@specialized T](a: Array[T],
-      fromIndex: Int, toIndex: Int, value: T): Unit = {
-    if (fromIndex > toIndex)
-      throw new IllegalArgumentException
-    if (fromIndex < 0 || toIndex > a.length)
-      throw new ArrayIndexOutOfBoundsException
-
-    var i = fromIndex
-    while (i != toIndex) {
-      a(i) = value
-      i += 1
-    }
+  def sort[T <: AnyRef](array: Array[T], comparator: Comparator[_ >: T]): Unit = {
+    implicit val ord = new Ordering[T] {
+      def compare(x: T, y: T): Int = comparator.compare(x, y)
+    }.asInstanceOf[Ordering[AnyRef]]
+    sortImpl[AnyRef](array.asInstanceOf[Array[AnyRef]])
   }
 
-  def fill(a: Array[AnyRef], value: AnyRef): Unit = {
-    var i = 0
-    while (i < a.length) {
-      a(i) = value
-      i += 1
-    }
+  def sort[T <: AnyRef](array: Array[T], fromIndex: Int, toIndex: Int, comparator: Comparator[_ >: T]): Unit = {
+    implicit val ord = new Ordering[T] {
+      def compare(x: T, y: T): Int = comparator.compare(x, y)
+    }.asInstanceOf[Ordering[AnyRef]]
+    sortRangeImpl[AnyRef](array.asInstanceOf[Array[AnyRef]], fromIndex, toIndex,
+      java.lang.reflect.Array.newInstance(
+        array.getClass.getComponentType, _).asInstanceOf[Array[AnyRef]])
   }
 
-  def fill(a: Array[AnyRef],
-      fromIndex: Int, toIndex: Int, value: AnyRef): Unit = {
-    if (fromIndex > toIndex)
-      throw new IllegalArgumentException
-    if (fromIndex < 0 || toIndex > a.length)
-      throw new ArrayIndexOutOfBoundsException
+  @inline
+  def sortImpl[@specialized T: ClassTag: Ordering](a: Array[T]): Unit =
+    scala.util.Sorting.stableSort(a)
```

Comment:
What's the point of this helper? I'd recommend calling `stableSort` directly from where it's called instead.